src/MDMSpringTimingCurve.m

```diff
@@ -41,5 +68,58 @@ - (instancetype)initWithMass:(CGFloat)mass
   return self;
 }
 
+- (CGFloat)mass {
+  [self recalculateCoefficientsIfNeeded];
+  return _mass;
+}
+
+- (CGFloat)tension {
+  [self recalculateCoefficientsIfNeeded];
+  return _tension;
+}
+
+- (CGFloat)friction {
+  [self recalculateCoefficientsIfNeeded];
+  return _friction;
+}
+
+#pragma mark - Private
+
+- (void)recalculateCoefficientsIfNeeded {
+  if (_coefficientsAreInvalid) {
+    [self recalculateCoefficients];
+  }
+}
+
+- (void)recalculateCoefficients {
+  UIView *view = [[UIView alloc] init];
+  [UIView animateWithDuration:_duration
+                        delay:0
+       usingSpringWithDamping:_dampingRatio
+        initialSpringVelocity:self.initialVelocity
+                      options:0
+                   animations:^{
+                     view.center = CGPointMake(100, 100);
+                   } completion:nil];
+
+  NSString *animationKey = [view.layer.animationKeys firstObject];
+  NSAssert(animationKey != nil, @"Unable to extract animation timing curve: no animation found.");
+#pragma clang diagnostic push
+  // CASpringAnimation is a private API on iOS 8 - we're able to make use of it because we're
+  // linking against the public API on iOS 9+.
+#pragma clang diagnostic ignored "-Wpartial-availability"
+  CASpringAnimation *springAnimation =
+  (CASpringAnimation *)[view.layer animationForKey:animationKey];
+  NSAssert([springAnimation isKindOfClass:[CASpringAnimation class]],
+           @"Unable to extract animation timing curve: unexpected animation type.");
+#pragma clang diagnostic pop
+
+  _mass = springAnimation.mass;
+  _tension = springAnimation.stiffness;
+  _friction = springAnimation.damping;
```

Comment:
Is there a reason these aren't static so that you only compute them once per app startup?

Reply:
Oh because they depend on the local vars ignore me

Reply:
The UIKit damping coefficient math is a function that looks roughly like this:

```
// Pseudo-code:
(mass, stiffness, damping) = coefficients(dampingRatio, duration, initialVelocity)
```

We could compute the values once per app startup is if we cache the results for the set of dampingRatio/duration/initialVelocity as inputs, but I'm not confident the added complexity is worth it here.

What do you think?

Reply:
Yeah you're right. Mmmmmaybe if MDMSpringTimingCurve has some defaults we could cache the results for the defaults, but otherwise not worth the effort.